src/expr/src/scalar/func.rs

```diff
@@ -6643,11 +6717,35 @@ impl VariadicFunc {
     /// introduces nulls even when it does not.
     pub fn introduces_nulls(&self) -> bool {
         use VariadicFunc::*;
-        // Note the negation
-        !matches!(
-            self,
-            And | Or //todo: add more
-        )
+        match self {
+            Concat
+            | PadLeading
+            | Substr
+            | ArrayCreate { .. }
+            | ListCreate { .. }
+            | RecordCreate { .. }
+            | ListSliceLinear
+            | SplitPart
+            | HmacString
+            | HmacBytes
+            | ErrorIfNull
+            | DateBinTimestamp
+            | DateBinTimestampTz
+            | RangeCreate { .. }
+            | And
+            | Or => false,
+            Coalesce
+            | Greatest
+            | Least
+            | MakeTimestamp
+            | Replace
+            | JsonbBuildArray
+            | JsonbBuildObject
+            | ArrayToString { .. }
+            | ArrayIndex { .. }
+            | ListIndex
+            | RegexpMatch => true,
```

Comment:
There might be a similar issue as with `IsRegexpMatch` above.

Reply:
`RegexpMatch` is implemented by `regexp_match_dynamic` which defers to `regexp_match_static` which returns null when there are no valid captures/no match. I don't 100% understand the way row packing works... if it packs just one null, will that be interpreted as a singleton array of nulls or a null proper?

Reply:
Ah, yes! It can indeed return null if there is no match. The `None => packer.push(Datum::Null),` does this, and there is no array created in this case. (The `Some` case uses `push_array` to construct an array (with possibly nulls inside from the inner `None` case).)

Reply:
Makes sense... thanks!